classes/output/report_table.php

```diff
@@ -0,0 +1,178 @@
+<?php
+// This file is part of Moodle - http://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+/**
+ * @package     local_smartmedia
+ * @author      Tom Dickman <tomdickman@catalyst-au.net>
+ * @copyright   2019 Catalyst IT Australia {@link http://www.catalyst-au.net}
+ * @license     http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+
+namespace local_smartmedia\output;
+
+defined('MOODLE_INTERNAL') || die;
+
+use table_sql;
+use moodle_url;
+use renderable;
+use html_writer;
+
+/**
+ * Renderable class for index page of local_smartmedia plugin.
+ *
+ * @package     local_smartmedia
+ * @author      Tom Dickman <tomdickman@catalyst-au.net>
+ * @copyright   2019 Catalyst IT Australia {@link http://www.catalyst-au.net}
+ * @license     http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+
+class report_table extends table_sql implements renderable {
+
+    /*
+     * Fields for query to populate table data.
+     */
+    const FIELDS = 'DISTINCT d.id ,f.filepath, f.filename, d.metadata, d.duration, d.size, d.width, d.height';
+
+    /*
+     * From statement for query to populate table data.
+     */
+    const FROM = "{local_smartmedia_data} d "
+                . "LEFT JOIN {files} f "
+                . "ON (d.contenthash = f.contenthash)";
+
+    /**
+     * Minimum width above which transcoding is considered HD.
+     */
+    const MIN_HD_WIDTH = 720;
+
+    /**
+     * report_table constructor.
+     *
+     * @param string $uniqueid Unique id of table.
+     * @param \moodle_url $url Url where this table is displayed.
+     * @param \stdClass $params the parameters for table display.
+     *
+     * @throws \coding_exception
+     */
+    public function __construct($uniqueid, $url, $params) {
+        parent::__construct($uniqueid);
+
+        $this->set_attribute('id', 'local_smartmedia_report_table');
+        $this->set_attribute('class', 'generaltable generalbox');
+        $this->show_download_buttons_at(array(TABLE_P_BOTTOM));
+        $this->is_downloading($params->download, 'smartmedia-report');
+        $this->define_baseurl($url);
+        $this->define_columns(array('filename', 'type', 'width', 'duration', 'size', 'cost'));
```

Comment:
Remove filename here, replace with file type. Where the options are Video or Audio. You can detmine this from the number of streams in the metadata table. Where a video file has >0 video streams and an audio file has video streams =0 and audio streams > 0.

Change existing type field to "format" instead.

Reply:
Replaced with a file type check for audio or video